config/typescript.js

```diff
@@ -1,20 +1,79 @@
 module.exports = {
   parser: '@typescript-eslint/parser',
+  parserOptions: {
+    'sourceType': 'module',
+  },
   plugins: [
     '@typescript-eslint',
   ],
-  extends: [
-    'eslint:recommended',
-    'plugin:@typescript-eslint/eslint-recommended',
-    'plugin:@typescript-eslint/recommended',
-  ],
   rules: {
+    // Checked by TypeScript - ts(2378)
+    'getter-return': 'off',
+    // Checked by TypeScript - ts(2300)
+    'no-dupe-args': 'off',
+    // Checked by TypeScript - ts(1117)
+    'no-dupe-keys': 'off',
+    // Checked by TypeScript - ts(7027)
+    'no-unreachable': 'off',
+    // Checked by TypeScript - ts(2367)
+    'valid-typeof': 'off',
+    // Checked by TypeScript - ts(2588)
+    'no-const-assign': 'off',
+    // Checked by TypeScript - ts(2588)
+    'no-new-symbol': 'off',
+    // Checked by TypeScript - ts(2376)
+    'no-this-before-super': 'off',
+    // Checked by TypeScript - `strictNullChecks`
+    'no-undef': 'off',
+    // Checked by TypeScript
+    'no-dupe-class-members': 'off',
+    // Checked by TypeScript
+    'no-redeclare': 'off',
+    // Checked by TypeScript
+    'import/no-unresolved': 'off',
```

Comment:
This is to say, the TS compiler takes care of these?

Reply:
Yup, the compiler should check all of these